src/meta/src/manager/event_log.rs

```diff
@@ -0,0 +1,288 @@
+    enabled: bool,
+    lease_ttl_sec: u64,
+    flush_interval_ms: u64,
+    max_payload_size: u64,
+) -> EventLogManger {
+    use futures::FutureExt;
+    use prost::Message;
+    let (event_tx, mut event_rx) = tokio::sync::mpsc::channel::<EventLog>(BUFFER_SIZE);
+    let (shutdown_tx, shutdown_rx) = tokio::sync::oneshot::channel();
+    let meta_store_clone = meta_store.clone();
+    let mut last_lease = None;
+    let shutdown_rx_shared = shutdown_rx.shared();
+    let worker_loop = async move {
+        let mut flush_interval = tokio::time::interval(Duration::from_millis(flush_interval_ms));
+        flush_interval.set_missed_tick_behavior(tokio::time::MissedTickBehavior::Delay);
+        flush_interval.reset();
+        let client = match as_etcd_client(&meta_store_clone) {
+            None => return,
+            Some(c) => c,
+        };
+        let mut event_log_batch: Vec<EventLog> = vec![];
+        loop {
+            futures::select_biased! {
+                _ = shutdown_rx_shared.clone().fuse() => {
+                    tracing::info!("event log worker is stopped");
+                    return;
+                },
+                _ = flush_interval.tick().fuse() => {
+                    // Flush batched event logs to persistent storage.
+                    if event_log_batch.is_empty() {
+                        continue;
+                    }
+                    last_lease = match may_get_new_lease(client, last_lease, lease_ttl_sec).await {
+                        Ok(last_lease) => last_lease,
+                        Err(e) => {
+                            tracing::warn!("Some event logs are not persisted: {}", e);
+                            continue;
+                        }
+                    };
+                    if let Err(e) = add(
+                        client,
+                        event_log_batch.drain(..).map(|e|(e.unique_id.to_owned().unwrap().as_bytes().into(),e.encode_to_vec())),
+                        last_lease.as_ref().unwrap().0,
+                        max_payload_size,
+                    )
+                    .await
+                    {
+                        tracing::warn!("Some event logs are not persisted: {}", e);
+                        continue;
+                    }
+                }
+                event_log = event_rx.recv().fuse() => {
+                    // Batch event logs.
+                    match event_log {
+                        None => {
+                            break;
+                        }
+                        Some(event_log) => {
+                            event_log_batch.push(event_log);
+                        }
+                    };
+                },
+            }
+        }
+    };
+    let worker_join_handle = tokio::spawn(worker_loop);
+    EventLogManger::new(
+        meta_store,
+        event_tx,
+        (worker_join_handle, shutdown_tx),
+        enabled,
+    )
+}
+
+async fn may_get_new_lease(
```

Comment:
What is this function for, could you add some docstring?

Reply:
Fixed